mode/clike/clike.js

```diff
@@ -519,6 +519,16 @@ CodeMirror.defineMode("clike", function(config, parserConfig) {
         stream.eatWhile(/[\w\$_]/);
         return "meta";
       },
+      "{": function(stream, state) {
+        // match a single full line with params of multi-line anonymous fn, e.g.:
+        // { a =>\n
+        // { (aa: Int, bb: Int) =>\n
+        var singleLineParams = /^(\w|[:.,() ])+=>$/;
+        if (!stream.match(singleLineParams, true)) // eat the line, if matches
```

Comment:
> This'll disable all highlighting for the parameters. That seems undesirable.

sure, I'd be more than happy to keep it. but it's much better as is, than the horrible identation we had before.

> Also, I believe Scala allows default values for parameters

not for anonymous fns, this is too weird / wouldn't compile:
```scala
scala> Option(1).map {x: Int = 1 => 1}
<console>:1: error: ';' expected but '=' found.
       Option(1).map {x: Int = 1 => 1}
                             ^
```

> Would it make more sense to delay the context manipulation until reaching the =>?

I had this in mind (but I'm not codemirror guru either). so what is needed to preserve the indentation, is to delay this context manipulation until whole line was `eaten` (into `highlighted parameters`), right?